pandas/core/dtypes/missing.py

```diff
@@ -394,3 +394,10 @@ def na_value_for_dtype(dtype):
     elif is_bool_dtype(dtype):
         return False
     return np.nan
+
+
+def remove_na_arraylike(series):
+    """
+    Return series containing only true/non-NaN values, possibly empty.
+    """
```

Comment:
ok with tests. the name is fine. its descriptive and actually works on arrays as well. 

yes if you can throw in a couple of tests (with parametrize) would be great.